app/views/spree/orders/_line_item.html.haml

```diff
@@ -29,7 +33,10 @@
         :min => 0, "ofn-on-hand" => "#{variant.on_demand && 9999 || variant.on_hand}",
         "finalizedquantity" => finalized_quantity, :class => "line_item_quantity", :size => 5,
         "ng-model" => "line_item_#{line_item.id}",
-        "validate-stock-quantity" => true
+        "validate-stock-quantity" => true,
+        :oninput => "document.getElementById('message-element').innerHTML = ''; if (this.value == this.getAttribute('max')) { document.getElementById('message-element').innerHTML = '#{t(".insufficient_stock", :on_hand => variant.on_hand)}';}"
```

Comment:
Sure it would be more consistent but it doesn't work for me. As I said, I'm not expert with Angular so may be I am missing something (it should be simple). If it's a real problem for you I am ok to leave it for someone else.